src/App/Fossa/Analyze/Log4jReport.hs

```diff
@@ -0,0 +1,299 @@
+
+printProjectReport :: (Has Diag.Diagnostics sig m) => ProjectResult -> m (Doc AnsiStyle)
+printProjectReport ProjectResult{..} =
+  pure $
+    vsep
+      [ ""
+      , "Project Path: " <> (pretty . show $ projectResultPath)
+      , pretty $ Text.intercalate "" $ replicate (length $ "Project Path: " <> (show projectResultPath)) "-"
+      , listPretty "Project Manifest Files" (pretty . show <$> projectResultManifestFiles)
+      , listPretty "Direct Inclusions" (map withAnnotation directInclusion)
+      , listPretty "Indirect Inclusions" (map withAnnotation inDirectInclusion)
+      ]
+  where
+    listPretty :: Text -> [Doc AnsiStyle] -> Doc AnsiStyle
+    listPretty header items =
+      if null items
+        then (pretty (header <> ": ")) <> "0 log4j dependencies found!"
+        else vsep $ (pretty header <> ":") : map ("- " <>) items
+
+    directInclusion :: [Log4jVulnerableReportItem]
+    directInclusion = map (toVulnerableReportItem toVulnerableDependencyManifestFiles) onlyDirects
+
+    inDirectInclusion :: [Log4jVulnerableReportItem]
+    inDirectInclusion = map (toVulnerableReportItem toVulnerableDependencyRootDependency) allDeep
+
+    withAnnotation :: Log4jVulnerableReportItem -> Doc AnsiStyle
+    withAnnotation (Log4jVulnerableReportItem name version origin vulTypes) =
+      colorCoded $
+        Text.intercalate
+          " "
+          [ name
+          , formattedVersion
+          , formattedVulnerability
+          , toText . show $ origin
+          ]
+      where
+        colorCoded doc = if Set.null vulTypes then (pretty doc) else (annotateWarn $ pretty doc)
+        formattedVersion = maybe "N/A" ("v" <>) (verConstraintToRevision =<< version)
+        formattedVulnerability =
+          if Set.null vulTypes
+            then inBracket "N/A"
+            else inBracket $ Text.intercalate ", " (map (toText . show) $ Set.toList vulTypes)
+
+    onlyDirects :: [Dependency]
+    onlyDirects = filter isRelevantDep $ directList projectResultGraph
+
+    -- Note, in our model dependency can be direct AND deep at the same time.
+    -- Dependency is *deep* only, when it has at-least one predecessor.
+    allDeep :: [Dependency]
+    allDeep = filter isRelevantDep $ filter (hasPredecessors projectResultGraph) $ vertexList projectResultGraph
+
+    isRelevantDep :: Dependency -> Bool
+    isRelevantDep Dependency{..} =
+      any
+        (\vd -> dependencyName == vdName vd)
+        getVulnerableDeps
+
+    toVulnerableDependencyManifestFiles :: Dependency -> VulnerableDependencyOrigin
+    toVulnerableDependencyManifestFiles _ = VulnerableDependencyManifestFiles projectResultManifestFiles
+
+    toVulnerableDependencyRootDependency :: Dependency -> VulnerableDependencyOrigin
+    toVulnerableDependencyRootDependency dep = VulnerableDependencyRootDependency (getRootsOf projectResultGraph dep)
+
+    toVulnerableReportItem :: (Dependency -> VulnerableDependencyOrigin) -> Dependency -> Log4jVulnerableReportItem
```

Comment:
This is not _elegant_ but this addresses a case, where dependency originates is transitive dependency as well as itself is direct dependency.